import/webengine/plugin.cpp

```diff
@@ -0,0 +1,49 @@
+/****************************************************************************
+**
+** Copyright (C) 2016 Jolla Ltd.
+** Contact: Raine Makelainen <raine.makelainen@jolla.com>
+**
+****************************************************************************/
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "webengine.h"
+#include "webenginesettings.h"
+
+#include <QtCore/QStandardPaths>>
+#include <QQmlExtensionPlugin>
+#include <qqml.h>
+#include <QtQml/QQmlEngine>
+#include <QtQml/QQmlContext>
+
+template <typename T> static QObject *singletonApiFactory(QQmlEngine *engine, QJSEngine *)
+{
+    return new T(engine);
+}
+
+class SailfishOSWebEnginePlugin : public QQmlExtensionPlugin
+{
+    Q_OBJECT
+    Q_PLUGIN_METADATA(IID "Sailfish.WebEngine" )
+
+public:
+    void initializeEngine(QQmlEngine *, const char *)
+    {
+        // TODO : How to deal with custom default UA. We have also means to customize
+        // site specific UA overrides.
+        SailfishOS::WebEngine::initialize(QStandardPaths::writableLocation(QStandardPaths::CacheLocation));
+        SailfishOS::WebEngineSettings::initialize();
```

Comment:
Let's figure out once we have more use cases that how should we actually do the initialization. Maybe this is something that should not happen here at all rather we could have explicit QML API. On the other hand, this kind of approach would be quite convenient.